src/cargo/util/command_prelude.rs

```diff
@@ -239,6 +239,14 @@ pub trait AppExt: Sized {
     fn arg_quiet(self) -> Self {
         self._arg(opt("quiet", "Do not print cargo log messages").short('q'))
     }
+
+    fn arg_timings(self) -> Self {
+        self._arg(optional_multi_opt(
```

Comment:
Clap question here. I don't really know how clap handles "optional payload" flags. For example how does this pase:

```
cargo test --timings arg
```

If you were to use `cargo test arg` then `arg` would filter what tests are run (it's passed to each test binary). If you run `cargo test --timings` it's clearly asking for timing information. Ideally my personal preference is that all flags which take optional values require a `=` to be used, disambiguating the above with `cargo test --timings=arg` for one interpretation and `cargo test --timings arg` is the other (where `arg` is a filter for tests).

Do you know whether we force one way or another within Cargo?

Reply:
@alexcrichton As far as I can tell, we handle this potential ambiguity in cargo by *never having positional arguments*. If we added any kind of positional argument, many of our "optional argument" options would break.

Reply:
I think the ship has sailed on that though? `cargo test arg` is valid today

Reply:
@alexcrichton Ah, I missed that when trying to look through Cargo commands for anything that took a positional argument.

Then I think that's an ambiguity, and an existing one we already have with other options that take an optional value.

Reply:
Hm I don't think that this is a preexisting bug, `optional_multi_opt` is only used for things like `--test` which IIRC is purely used for error reporting so if you do `cargo test --bin` then Cargo will print all the binaries or something like that. I don't think that there are any other instances of flags with optional values in Cargo.

Reply:
@alexcrichton I've updated the timings option to use `.require_equals(true)`, which makes it require an `=`. I've also modified it to use `optional_opt` rather than `optional_multi_opt`, since it doesn't seem critical for `--timings=html --timings=json` to work. (I can easily revert the latter if you prefer.)